tests/fv/fv_test.go

```diff
@@ -85,52 +86,112 @@ var _ = Describe("PolicyController", func() {
 		time.Sleep(time.Second * 15)
 	})
 
-	It("should create profiles in etcd to represent k8s namespaces", func() {
-		// Create the namespace in Kubernetes
-		ns := &v1.Namespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "peanutbutter",
-			},
-			Spec: v1.NamespaceSpec{},
-		}
-		_, err := k8sClient.CoreV1().Namespaces().Create(ns)
-		Expect(err).NotTo(HaveOccurred())
-
-		// Check that the matching profile was created in etcd.
-		Eventually(func() error {
-			_, err = calicoClient.Profiles().Get(api.ProfileMetadata{Name: "k8s_ns.peanutbutter"})
-			return err
-		}).ShouldNot(HaveOccurred())
-	})
-
-	Context("when etcd data is lost", func() {
+	Context("profiles", func() {
+		var profName string
 		BeforeEach(func() {
-			// Write some data
+			nsName := "peanutbutter"
+			profName = "k8s_ns." + nsName
 			ns := &v1.Namespace{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: "peanutbutter",
+					Name: nsName,
+					Labels: map[string]string{
+						"peanut": "butter",
+					},
 				},
 				Spec: v1.NamespaceSpec{},
 			}
 			_, err := k8sClient.CoreV1().Namespaces().Create(ns)
 			Expect(err).NotTo(HaveOccurred())
 
-			// Wait for it to appear in etcd
 			Eventually(func() *api.Profile {
 				profile, _ := calicoClient.Profiles().Get(api.ProfileMetadata{Name: "k8s_ns.peanutbutter"})
 				return profile
 			}).ShouldNot(BeNil())
 		})
-		It("should recover by rewriting the data to etcd ", func() {
-			Stop(apiserver)
-			err := calicoClient.Profiles().Delete(api.ProfileMetadata{Name: "k8s_ns.peanutbutter"})
+
+		It("should write new profiles in etcd to match namespaces in k8s ", func() {
+			err := calicoClient.Profiles().Delete(api.ProfileMetadata{Name: profName})
+			Expect(err).ShouldNot(HaveOccurred())
+			Eventually(func() error {
+				_, err := calicoClient.Profiles().Get(api.ProfileMetadata{Name: profName})
+				return err
+			}, time.Second*15, 500*time.Millisecond).ShouldNot(HaveOccurred())
+		})
+		It("should update existing profiles in etcd to match namespaces in k8s", func() {
+			profile, err := calicoClient.Profiles().Update(&api.Profile{Metadata: api.ProfileMetadata{Name: profName}})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(profile.Metadata.Labels).To(BeNil())
+			Eventually(func() map[string]string {
+				prof, _ := calicoClient.Profiles().Get(api.ProfileMetadata{Name: profName})
+				return prof.Metadata.Labels
+			}, time.Second*15, 500*time.Millisecond).ShouldNot(BeEmpty())
+		})
+	})
+
+	Context("policies", func() {
+		var policyName string
+		var genPolicyName string
+
+		BeforeEach(func() {
```

Comment:
A bit weird for this to be a BeforeEach, especially given there is nothing cleaning it up after each test AFAICT.